atomic_reactor/plugins/pre_reactor_config.py

```diff
@@ -37,6 +37,219 @@ def get_config(workflow):
         return conf
 
 
+def get_koji(workflow):
+    config = get_config(workflow)
+    if 'koji' in config:
+        return config['koji']
+
+
+def get_koji_session(workflow):
+    config = get_koji(workflow)
+
+    if not config:
+        return
+
+    from atomic_reactor.koji_util import create_koji_session
+    auth_info = {
+        "proxyuser": config['auth']['proxyuser'],
+        "ssl_certs_dir": config['auth']['ssl_certs_dir'],
+        "krb_principal": config['auth']['krb_principal'],
+        "krb_keytab": config['auth']['krb_keytab_path']
+    }
+    return create_koji_session(config['hub_url'], auth_info)
+
+
+def get_pulp(workflow):
+    config = get_config(workflow)
+    if 'pulp' in config:
+        return config['pulp']
```

Comment:
If lots of these functions will have almost identical implementation, there must be a way to avoid the duplication.

Reply:
I think we want to have all functions as it is more readable, then setattr on module, which we wanted avoid in the past